### PR TITLE
Split up encryption material into two flags

### LINKED ISSUE #4242: Separate handling of encryption inputs based on type
We should change from the current handling of `SINGULARITY_ENCRYPTION_KEY` that allows for both key file paths and passphrases to be specified in the same variable. 

This should be split into:
- `SINGULARITY_ENCRYPTION_PASSPHRASE` and `--passphrase`(that creates an interactive prompt) for encryption using a passphrase.
- `SINGULARITY_ENCRYPTION_PEM_PATH`/`--pem-path` for encryption using a public/private rsa key pair in the `pem` format.

### PR DESCRIPTION
**Pull request by ikaneshiro**
This includes changes to the 'crypt' package to accept a struct in its
external functions instead of a URI to avoid ambiguity for passphrases
that may potentially container a '://' and be incorrectly parsed

Signed-off-by: Ian Kaneshiro <iankane@umich.edu>

**Description of the Pull Request (PR):**

Write your description of the PR here. Be sure to include as much background,
and details necessary for the reviewers to understand exactly what this is
fixing or enhancing.

**This fixes or addresses the following GitHub issues:**

- Fixes #4242 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: cmd/internal/cli/action_flags.go [hunk 1]
```diff
@@ -295,14 +296,23 @@ var actionContainLibsFlag = cmdline.Flag{
...
+// --pem-path
+var actionPEMPathFlag = cmdline.Flag{
+	ID:           "actionEncryptionPEMPath",
+	Value:        &encryptionPEMPath,
 	DefaultValue: "",
-	Name:         "encryption-key",
-	Hidden:       true,
-	EnvKeys:      []string{"ENCRYPTION_KEY"},
+	Name:         "pem-path",
+	Usage:        "Enter an path to a PEM formated PSA key for an encrypted container",
```
Comment (gvallee) [inline cmd/internal/cli/action_flags.go]:
```suggestion
	Usage:        "Enter an path to a PEM formatted RSA key for an encrypted container",
```

File: cmd/internal/cli/build_linux.go [hunk 2]
```diff
@@ -315,3 +327,48 @@ func handleBuildFlags(cmd *cobra.Command) {
 		}
 	}
 }
+
+// this function handles the setting of encryption environment and flag parameters to eventually be
```
Comment (gvallee) [inline cmd/internal/cli/build_linux.go]:
```suggestion
// getEncryptionMaterial handles the setting of encryption environment and flag parameters to eventually be
```

**Review by gvallee — CHANGES_REQUESTED** [review]
Only a few minor things

**Review by gvallee — APPROVED** [review]
lgtm

**Comment by ikaneshiro** [comment]
Merging now so @gvallee can get e2e tests up soon. Will make any requested changes in a follow up pr

**Pull request merged**